src/operators.jl

```diff
@@ -183,8 +187,50 @@ end
 function *{T<:Real}(g::DiGraph, v::Vector{T})
     length(v) == nv(g) || error("Vector size must equal number of vertices")
     y = zeros(T, nv(g))
-    for (i,j) in edges(g)
+    for e in edges(g)
+        i = src(e)
+        j = dst(e)
         y[i] += v[j]
     end
     return y
 end
+
+"""sum(g,i) provides 1:indegree or 2:outdegree vectors"""
+function sum(g::Graph, dim::Int)
+    if dim == 1
+        return indegree(g, vertices(g))
+    elseif dim == 2
+        return outdegree(g, vertices(g))
+    else
+        return Error("Graphs are only two dimensional")
+    end
+end
+
+
+size(g::Graph) = (nv(g), nv(g))
+"""size(g,i) provides 1:nv or 2:nv else 1 """
+function size(g::Graph, dim::Int)
+    if dim == 1
```

Comment:
just curious: why not just `size(g::Graph,dim::Int) = (dim == 1 || dim == 2)? nv(g) : 1`